lib/iterator.js

```diff
@@ -204,7 +204,8 @@ function byKey (a, b, reverse) {
 
 function allDeletes (nodes) {
   for (var i = 0; i < nodes.length; i++) {
-    if (nodes[i].value !== null) return false
+    if (nodes[i].value === null || nodes[i].deleted) continue
```

Comment:
So we're saying that `null` is still going to represent a deletion (as in, even with this addition one wouldn't be able to have meaningful `null` values)?

Reply:
Definitely necessary for backwards-compat, but is that a desirable limitation once this change is ancient history?

Reply:
No I'd like to get rid of that of check later on. Should we just drop support for `null === deleted` in 3.0.0 stable? I'm okay doing that

Reply:
Yeah especially since 3.0.0 is considered an "LTS" release right?

Reply:
yes, i'll update that